ground/src/main/java/com/google/android/ground/persistence/sync/SurveySyncWorker.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.persistence.sync
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.Data
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.google.android.ground.repository.SurveyRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import timber.log.Timber
+
+/** Worker responsible for syncing latest surveys and LOIs from remote server to local db. */
+@HiltWorker
+class SurveySyncWorker
+@AssistedInject
+constructor(
+  @Assisted context: Context,
+  @Assisted params: WorkerParameters,
+  private val surveyRepository: SurveyRepository
+) : Worker(context, params) {
+  private val surveyId: String? = params.inputData.getString(SURVEY_ID_PARAM_KEY)!!
+
+  override fun doWork(): Result {
+    if (surveyId == null) {
+      Timber.e("Survey sync scheduled with null surveyId")
+      return Result.failure()
+    }
+
+    // It's ok to block here since WorkManager calls doWork() on a background thread.
+    Timber.d("Syncing survey $surveyId")
+    surveyRepository.syncSurveyWithRemote(surveyId).ignoreElement().blockingAwait()
```

Comment:
Out of curiosity, is there a single background thread for the entire application or does each worker spawn in a new thread?

Reply:
The docs are conspicuously silent on this question:

> By default, WorkManager sets up an Executor for you—but you can also customize your own. 
https://developer.android.com/guide/background/persistent/threading/worker

How can we verify this easily?

Reply:
Found it:

```java
private @NonNull Executor createDefaultExecutor() {
  return Executors.newFixedThreadPool(
          // This value is the same as the core pool size for AsyncTask#THREAD_POOL_EXECUTOR.
          Math.max(2, Math.min(Runtime.getRuntime().availableProcessors() - 1, 4)));
}
```
https://android.googlesource.com/platform/frameworks/support.git/+/a9ac247af2afd4115c3eb6d16c05bc92737d6305/work/workmanager/src/main/java/androidx/work/Configuration.java#107

Reply:
Thanks for confirming